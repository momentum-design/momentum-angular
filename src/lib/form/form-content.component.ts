/** @component form */

import { Component, Input } from '@angular/core';

/**
 * FormContent helps organize the content within a form section and provides a S wrapper;
 **/

@Component({
  selector: 'md-form-content',
  template: `
    <ng-content></ng-content>
  `,
  host: {
    'class': 'section__content',
  }
})

export class FormContentComponent {}
