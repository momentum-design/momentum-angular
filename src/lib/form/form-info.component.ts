/** @component form */

import { Component, Input } from '@angular/core';

/**
 * FormInfo is supplemental, organizational component used to help divide form
 **/

@Component({
  selector: 'md-form-info',
  template: `
    <h4 class="section__title">{{ title }}</h4>
    <p *ngIf="description" class="section__description">{{ description }}</p>
  `,
  host: {
    'class': 'section__info',
  }
})

export class FormInfoComponent {
  /** @prop Optional FormInfo description text | '' */
  @Input() description: string = '';
  /** @prop Optional FormInfo title | '' */
  @Input() title: string = '';
}
