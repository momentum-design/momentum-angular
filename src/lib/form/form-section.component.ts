/** @component form */

import { Component, Input } from '@angular/core';

@Component({
  selector: 'md-form-section',
  template: `
    <md-form-info [title]="title" [description]="description"></md-form-info>
    <md-form-content><ng-content></ng-content></md-form-content>
  `,
  host: {
    'class': 'md-form__section',
  }
})

export class FormSectionComponent {
  /** @prop Optional FormSection description | '' */
  @Input() description: string = '';
  /** @prop Optional FormSection title | null */
  @Input() title: string;
}
