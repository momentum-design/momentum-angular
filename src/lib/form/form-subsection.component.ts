/** @component form */

import { Component, Input } from '@angular/core';

@Component({
  selector: 'md-form-sub-section',
  template: `
    <h5 *ngIf="label" class="sub-section__label">{{ label }}</h5>
    <p *ngIf="description" class="sub-section__description">{{ description }}</p>
    <ng-content></ng-content>
  `,
  host: {
    'class': 'sub-section',
  }
})

export class FormSubSectionComponent {
  /** @prop Optional FormSubSection description text | '' */
  @Input() description: string = '';
  /** @prop Optional FormSubSection label text | '' */
  @Input() label: string = '';
}
