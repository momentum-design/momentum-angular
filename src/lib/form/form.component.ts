/** @component form */

import { Component, Input } from '@angular/core';

@Component({
  selector: 'md-form',
  template: `
    <ng-content></ng-content>
  `,
  host: {
    'class': 'md-form',
    '[attr.name]': 'name',
  }
})

export class FormComponent {
  /** @prop Form name */
  @Input() name: string;
}
