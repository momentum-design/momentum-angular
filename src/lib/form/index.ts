export * from './form.module';
export * from './form.component';
export * from './form-content.component';
export * from './form-info.component';
export * from './form-section.component';
export * from './form-subsection.component';

