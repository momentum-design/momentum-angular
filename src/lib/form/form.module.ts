import { CommonModule } from '@angular/common';
import { NgModule, CUSTOM_ELEMENTS_SCHEMA, ModuleWithProviders } from '@angular/core';
import { FormComponent } from './form.component';
import { FormContentComponent } from './form-content.component';
import { FormInfoComponent } from './form-info.component';
import { FormSectionComponent } from './form-section.component';
import { FormSubSectionComponent } from './form-subsection.component';


@NgModule({
  imports: [
    CommonModule,
  ],
  declarations: [
    FormComponent,
    FormContentComponent,
    FormInfoComponent,
    FormSectionComponent,
    FormSubSectionComponent,
  ],
  exports: [
    FormComponent,
    FormContentComponent,
    FormInfoComponent,
    FormSectionComponent,
    FormSubSectionComponent,
  ],
  schemas: [CUSTOM_ELEMENTS_SCHEMA],
})
export class FormModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: FormModule,
      providers: [],
    };
  }
}
