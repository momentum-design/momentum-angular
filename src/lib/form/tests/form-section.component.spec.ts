import { Component } from '@angular/core';
import { async, TestBed } from '@angular/core/testing';
import { FormSectionComponent } from '../form-section.component';
import { FormModule } from '../form.module';

describe('FormSectionComponent', () => {
  @Component({
    selector: 'test-app',
    template: `
      <md-form-section title="test">
        <div class="testingChildren"></div>
        <div class="testingChildren"></div>
      </md-form-section>
    `,
  })
  class TestAppComponent {}

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormModule],
      declarations: [TestAppComponent],
    }).compileComponents();
  }));

  it('should create and match snapshot', () => {
    const fixture = TestBed.createComponent(FormSectionComponent);
    const testComponent = fixture.componentInstance;
    fixture.detectChanges();
    expect(testComponent).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render one FormSection', () => {
    const fixture = TestBed.createComponent(FormSectionComponent);
    const testComponent = fixture.componentInstance;
    testComponent.title = 'test';
    fixture.detectChanges();
    expect(fixture.nativeElement.className).toContain('md-form__section');
    expect(fixture.nativeElement.querySelector('.section__title')).not.toBeNull();
    expect(fixture.nativeElement.querySelector('.section__description')).toBeNull();
  });

  it('should render description', () => {
    const fixture = TestBed.createComponent(FormSectionComponent);
    const testComponent = fixture.componentInstance;
    testComponent.description = 'test';
    testComponent.title = 'test';
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.section__description');
    expect(element.textContent).toEqual('test');
  });

  it('should render children under Form Content wrapper', () => {
    const fixture = TestBed.createComponent(TestAppComponent);
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelectorAll('.testingChildren');
    expect(element.length).toEqual(2);
    const element2 = fixture.nativeElement.querySelector('md-form-content');
    expect(element2.children.length).toEqual(2);
  });
});
