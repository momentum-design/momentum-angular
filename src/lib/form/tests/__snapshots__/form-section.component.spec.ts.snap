// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSectionComponent should create and match snapshot 1`] = `
<md-form-section
  description=""
>
  <md-form-info
    class="section__info"
    ng-reflect-description=""
  >
    <h4
      class="section__title"
    >
      
    </h4>
    
  </md-form-info><md-form-content
    class="section__content"
  />
</md-form-section>
`;
